doc/source/dev_guide/satpy_internals.rst

```diff
@@ -0,0 +1,98 @@
+======================================================
+ Satpy internal workings: having a look under the hood
+======================================================
+
+Querying and identifying data arrays
+====================================
+
+DataQuery's
```

Comment:
Use of ' for plural is not correct in English, dropping it would be confusing, so I would make the heading just `DataQuery`.